pp.h

```diff
@@ -701,14 +701,7 @@ True if this op will be the return value of an lvalue subroutine
 =cut */
 #define LVRET ((PL_op->op_private & OPpMAYBE_LVSUB) && is_lvalue_sub())
 
-#define SvCANEXISTDELETE(sv) \
- (!SvRMAGICAL(sv)            \
-  || !(mg = mg_find((const SV *) sv, PERL_MAGIC_tied))           \
-  || (   (stash = SvSTASH(SvRV(SvTIED_obj(MUTABLE_SV(sv), mg)))) \
-      && gv_fetchmethod_autoload(stash, "EXISTS", TRUE)          \
-      && gv_fetchmethod_autoload(stash, "DELETE", TRUE)          \
-     )                       \
-  )
+#define SvCANEXISTDELETE(sv) Perl_sv_can_existdelete(aTHX_ (SV *)sv)
```

Comment:
Why the plain cast here? Shouldn't this be `MUTABLE_SV(sv)`?

Reply:
I've never really worked out why those macros exist, but sure I guess for consistency ;)

Reply:
It's so you get a warning if you cast away `const`, at least on gcc/clang. Basically, you're asserting that an expression has a pointer-to-non-const type. (Which is why it's not needed in the inline function because there it's just a local variable whose declaration you can look at.)